components/TextFields/src/MDCInputTextField.h

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+/**
+ A set of Contained Input View states outlined in the Material guidelines. These states overlap with
+ and extend UIControlState.
+ */
+typedef NS_OPTIONS(NSInteger, MDCContainedInputViewState) {
```

Comment:
I feel like this should be `MDCInputTextFieldState` what do you think?

Reply:
I agree that `MDCContainedInputViewState` is kind of a silly name, since `MDCContainedInputView` is now private. However, I want to build several more "contained input views in disguise" (like the input chip view and the text area) and IMO it makes sense for them all to share the same type for state. I see two options:
- A type that accounts for the states for all these input views, but only these input views 
- A type that accounts for the Material states across our components, which includes all the cases these input views require.

Reply:
Wouldn't all of those "contained input views in disguise" require "input", maybe `MDCInputFieldState`? What do you think?

Reply:
Not sure